docs/src/main/sphinx/client/jdbc.md

```diff
@@ -246,3 +246,11 @@ may not be specified using both methods.
   - Sets the time zone for the session using the [time zone
     passed](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/ZoneId.html#of(java.lang.String)).
     Defaults to the timezone of the JVM running the JDBC driver.
+* - `legacyPreparedStatements`
+  - Defaults to `true`. When set to `false`, prepared statements are executed
+    calling a single `EXECUTE IMMEDIATE` query instead of the standard
+    `PREPARE <statement>` followed by `EXECUTE <statement>`. This reduces
+    network overhead and uses smaller HTTP headers but requires Trino 431 or
+    greater. Note that calling `PreparedStatement.getMetadata()` or
```

Comment:
Not sure we want this code level info .. maybe .. given its the JDBC driver. What do you think @electrum ?

Reply:
I would still like to get an opinion from @electrum to see if this kind of Java code level doc is desired for the JDBC driver docs. I can see arguments for yes and no..